spec/system/user_page/staff_info_spec.rb

```diff
@@ -17,4 +17,45 @@
       expect(user_page).to have_warning_messages_path(user)
     end
   end
+
+  context "for flagged posts" do
+    before do
+      ReviewableFlaggedPost
+        .statuses
+        .except(:approved)
+        .keys
+        .map do |status|
+          Fabricate(:reviewable_flagged_post, status: status, target_created_by: user)
+        end
+    end
+
+    context "when there are no approved flagged posts" do
+      it "should not display a flagged-posts staff counter" do
+        expect(user_page.visit(user).staff_info_flagged_posts_counter).to be_nil
+      end
+    end
+
+    context "when there are approved flagged posts" do
+      before do
+        2.times do
+          PostActionCreator
+            .off_topic(admin, Fabricate(:post, user: user))
+            .reviewable
+            .perform(admin, :agree_and_keep)
+        end
+      end
+
+      it "should show the right count in the flagged-posts staff counter" do
+        user_page.visit(user)
+        expect(
+          user_page.staff_info_flagged_posts_counter.text.to_i,
+        ).to eq ReviewableFlaggedPost.approved.count
```

Comment:
In system tests, we're sort of treating the app as a black box so I don't think the assertions should be "aware" of the models in the app. In this case, I think an assertion like this will be clearer:

`expect(user_page).to have_staff_info_flagged_posts_count(count: 2)`